core/src/main/java/org/infinispan/cache/impl/CacheImpl.java

```diff
@@ -557,6 +558,14 @@ private V removeInternal(Object key, EnumSet<Flag> explicitFlags, InvocationCont
       return (V) executeCommandAndCommitIfNeeded(ctx, command);
    }
 
+   @Override
+   public void removeExpired(K key, V value, Long lifespan) {
+      InvocationContext ctx = getInvocationContextWithImplicitTransaction(false, null, 1);
+      RemoveExpiredCommand command = commandsFactory.buildRemoveExpiredCommand(key, value, lifespan);
+      ctx.setLockOwner(command.getKeyLockOwner());
```

Comment:
The RemoveExpiredCommand is never done in an ongoing transaction or from a user operation.  The removal is done asynchronously when a get finds an expired value and the reaper thread is isolated.  Or is there something else here?